store/copr/coprocessor.go

```diff
@@ -1037,22 +1094,23 @@ func (worker *copIteratorWorker) handleTiDBSendReqErr(err error, task *copTask,
 	return nil
 }
 
-func (worker *copIteratorWorker) buildCopTasksFromRemain(bo *Backoffer, lastRange *coprocessor.KeyRange, task *copTask) ([]*copTask, error) {
+func (worker *copIteratorWorker) buildCopTasksFromRetry(bo *Backoffer, lastRange *coprocessor.KeyRange, task *copTask) ([]*copTask, error) {
 	remainedRanges := task.ranges
 	if worker.req.Streaming && lastRange != nil {
-		remainedRanges = worker.calculateRemain(task.ranges, lastRange, worker.req.Desc)
+		remainedRanges = worker.calculateRetry(task.ranges, lastRange, worker.req.Desc)
 	}
-	return buildCopTasks(bo, worker.store.GetRegionCache(), remainedRanges, worker.req, task.eventCb)
+	task.ranges = remainedRanges
+	return []*copTask{task}, nil
```

Comment:
IMO, rebuilding tasks by region cache seems redundant here, because this function is triggered by a lock error instead of a region error. Though it affects streaming only, I remove it.

Reply:
Why not remove this function and move this logic to L967?